src/lockTracer.cpp

```diff
@@ -64,16 +64,18 @@ Error LockTracer::start(Arguments& args) {
     return Error::OK;
 }
 
-void LockTracer::stop() {
+void LockTracer::stop(bool restart) {
     jvmtiEnv* jvmti = VM::jvmti();
     JNIEnv* env = VM::jni();
 
     // Disable Java Monitor events
     jvmti->SetEventNotificationMode(JVMTI_DISABLE, JVMTI_EVENT_MONITOR_CONTENDED_ENTER, NULL);
     jvmti->SetEventNotificationMode(JVMTI_DISABLE, JVMTI_EVENT_MONITOR_CONTENDED_ENTERED, NULL);
 
-    // Reset Unsafe.park() trap
-    setUnsafeParkEntry(env, _orig_unsafe_park);
+    if (!restart) {
+        // Reset Unsafe.park() trap
+        setUnsafeParkEntry(env, _orig_unsafe_park);
+    }
```

Comment:
After discussing this with @Baraa-Hasheesh, I start thinking this condition is not really correct/useful.
What if we don't reset the hook at all? In the worst case, we'll just have one extra call for each `Unsafe.park` invocation, which is a drop in the ocean comparing to all other stuff `Unsafe.park` is doing. This is a slow path anyway.